src/AWSCore.jl

```diff
@@ -551,15 +546,16 @@ function do_request(r::AWSRequest)
         catch e
             @ignore if typeof(e) == KeyError end
         end
-        return info
+
+        return (return_headers ? (info, Dict(response.headers)) : info)
     end
 
     if occursin(r"^text/", mime)
-        return body
+        return (return_headers ? (body, Dict(response.headers)) : body)
     end
 
     # Return raw data by default...
-    return response.body
+    return (return_headers ? (response.body, nothing) : response.body)
```

Comment:
Why `nothing` here? This is the codepath I'm seeing AWSS3 take with your changes, resulting in:

```
MethodError: no method matching getindex(::Nothing, ::String)
Stacktrace:
 [1] s3_upload_part(::Dict{Symbol,Any}, ::XMLDict.XMLDictElement, ::Int64, ::Array{UInt8,1}) at /home/keno/.julia/dev/AWSS3/src/AWSS3.jl:698
```

Reply:
Fixed in #117 , I think the comment above it threw me off.